oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/ClusterNodeInfo.java

```diff
@@ -229,13 +229,14 @@ static RecoverLockState fromString(String state) {
     /** OAK-3399 : max number of times we're doing a 1sec retry loop just before declaring lease failure **/
     private static final int MAX_RETRY_SLEEPS_BEFORE_LEASE_FAILURE = 5;
 
-    /** OAK-10281 : seconds to delay a recovery after a lease timeout */
-    private static final int DEFAULT_RECOVERY_DELAY_SECS = SystemPropertySupplier.create("oak.documentMK.recoveryDelaySecs", 0)
-            .loggingTo(LOG).validateWith(value -> value >= 0)
-            .formatSetMessage((name, value) -> String.format("recovery delay set to (secs): %ss (using system property %s)", name, value)).get();
-    private static final long DEFAULT_RECOVERY_DELAY_MILLIS = 1000L * (long)DEFAULT_RECOVERY_DELAY_SECS;
+    /** OAK-10281 : default millis to delay a recovery after a lease timeout */
+    static final long DEFAULT_RECOVERY_DELAY_MILLIS = 0;
 
-    // kept non-final for testing purpose
+    /**
+     * Actual millis to delay a recovery after a lease timeout.
+     * <p>
+     * Initialized by DocumentNodeStore constructor.
+     */
     static long recoveryDelayMillis = DEFAULT_RECOVERY_DELAY_MILLIS;
```

Comment:
I would make this field final and if needed would use reflection to override value in test cases.

See https://github.com/apache/jackrabbit-oak/blob/trunk/oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStoreTest.java#L434

Reply:
That would have been an option, but given I like to explicitly overwrite it in tests, I have now chosen the non final way. What's the advantage of having it final?

Reply:
Since this variable is a compile time constant, I would make it final and if required would use reflection (or any other work around) to update this in tests OR better write test as such that I don't need to override in the first place.

Reply:
Upon revisit, non-final actually doesn't work as this is modified for both test and non-test case. The latter being the main use case: [DocumentNodeStoreService.configureBuilder](https://github.com/apache/jackrabbit-oak/blob/OAK-10281-2/oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStoreService.java#L528).

Replacing this static field altogether, which would probably be preferred, however isn't possible without a larger change : it would somehow have to be passed to ClusterNodeInfoDocument for it to be available at isRecoveryNeeded invocation time. That could eg be done if it was available at constructor time. From there it goes on to perhaps being available in DocumentStore. But all of that doesn't make it simpler but rather adds more clutter... which is the reason for the static approach. Not the nicest, but somewhat nicer than adding it to several additional objects.

If we'd go the static route, then I'd argue it can't be final. And going the non-static route seems heavy.

Reply:
To me it seems to be an anti-pattern or hacky way of implementing this change.
IMO our approach shouldn't be decided by no. of changes rather than correct way of implementing it.

But in this case if you think the no. of changes are too much of an effort vs the gain that we get from them, then may be we can go with static field approach as well. I will leave that upto you to decide.

Reply:
> To me it seems to be an anti-pattern or hacky way of implementing this change.

If it is considered anti-pattern or hacky that is of course not good.

I'll work on an alternative

Reply:
Created a draft of an alternative at https://github.com/apache/jackrabbit-oak/pull/1292